### ISSUE TITLE
UI Grid dropdown empty space item display

### ISSUE THREAD

**Issue by ArunMNew** [angular-ui/ui-grid#3942]
Hi,

In UI Grid when i use "'ui-grid/dropdownEditor" editableCellTemplate for display the dropdown
by default it shows an extra empty space item as first index along with actual itemsource.

I want to  load actual itemsource without any selected item ...

Is there anyway to remove/hide that empty space item ?

![uigriddropdown](https://cloud.githubusercontent.com/assets/12839746/8541178/dbb0f2bc-24a9-11e5-944b-7697dfaa0dd0.jpg)

**Comment by kranthikumar2555** [comment]
that empty space item rendered in the html (<option value="" label=""></option> )
Temp Solution  is we can remove it by CSS trick like

#GridCustomId.ui-grid-cell div select option:first-child[value=""] {
        display: none;
    }

**Comment by ArunMNew** [comment]
@kranthikumar2555 , i have tried with your suggestion as below. But it did not resolve the problem.
What i am doing wrong here please ?

<html ng-app="app">
  <head>
    <script src="http://ajax.googleapis.com/ajax/libs/angularjs/1.3.16/angular.js"></script>
    <script src="http://ajax.googleapis.com/ajax/libs/angularjs/1.3.16/angular-touch.js"></script>
    <script src="http://ajax.googleapis.com/ajax/libs/angularjs/1.3.16/angular-animate.js"></script>
    <script src="http://ui-grid.info/docs/grunt-scripts/csv.js"></script>
    <script src="http://ui-grid.info/docs/grunt-scripts/pdfmake.js"></script>
    <script src="http://ui-grid.info/docs/grunt-scripts/vfs_fonts.js"></script>
    <script src="/release/ui-grid-unstable.js"></script>
    <script src="/release/ui-grid-unstable.css"></script>
    <script src="app.js"></script>
  
    <style>
      #gridOptions.ui-grid-cell div select option:first-child[value=""] {
display: none;
}
   <style>

  </head>
  <body>
    <div ng-controller="MainCtrl">
      <strong>Data Length:</strong> {{ gridOptions.data.length | number }}
      <br>
      <strong>Last Cell Edited:</strong> {{msg.lastCellEdited}}
      <br>
      <div ui-grid="gridOptions" ui-grid-edit class="grid"></div>
      <br>
      <div><strong>Last file uploaded:</strong></div>
      <div>{{lastFile}}</div>
    </div>
  </body>
</html>

**Comment by ArunMNew** [comment]
@kranthikumar2555 , i have tried your suggestion with below steps, but it does not help.

Please tell me what i am doing wrong here.

below is the html div which renders the ui grid.

&lt;div ui-grid="gridOptions" ui-grid-edit class="grid"&gt;&lt;/div&gt;

i have added style as below.
&lt;style&gt;
#gridOptions.ui-grid-cell div select option:first-child[value=""] {
display: none;
}
&lt;/style&gt;

**Comment by JLLeitschuh** [comment]
You need to set the **id** of the dom element you are using the directive on. In your case:
```
<div id="GridCustomId" ui-grid="gridOptions" ui-grid-edit class="grid"></div>
```
With the style:
```
<style>
  #GridCustomId.ui-grid-cell div select option:first-child[value=""] {
    display: none;
  }
</style>
```
Lookup how CSS selectors work in order to understand how `#GridCustomId.ui-grid-cell div select option:first-child[value=""]` is resolved.

**Comment by ArunMNew** [comment]
@JLLeitschuh , i created sample plunker and did the same above approach. Still issue was not resolved.

Please see  the plunker http://plnkr.co/edit/vhAATILeLpfd2CJONKi6?p=preview

In this plunker page the "Gender" column dropdown has empty space item.
I have added that style in "main.css" file.

**Comment by bernhard-hofmann** [comment]
The blank value will be removed when you chose one of the options. AngularJS adds this blank option when your model value is not in the options for the drop down list. If you want it to default to a value, you should set that on the object that is bound to that row in your grid.

**Comment by ArunMNew** [comment]
@bernhard-hofmann , so this is not possible to remove that blank option without setting to any default value ?

**Comment by bernhard-hofmann** [comment]
If there wasn't a blank, which value would you expect to be selected?

**Comment by aarosil** [comment]
@bernhard-hofmann  Should the above also apply if the choice was already set in the data?  I see blank value as an option, even though my data has a value set.  Is this because I am binding to object rather than ID ("field" prop of the columnDef points to an object).  Is there some "track by" I can use to indicate what option is already selected in the data (such as with ng-options)?

**Comment by PaulL1** [comment]
Often the trick is 1 v's "1" in your data.  i.e. string v's numeric.

**Comment by aarosil** [comment]
Thanks for the confirmation I should be able to bind to an object, display it's "name" prop and not have a "blank" dropdown option.  

I confirmed both my original data in the grid as well as the data in `editDropdownOptionsArray` both have numeric IDs.  

I thought maybe the blank option appeared because I assigned the dropdown values after async call, but I see the blank option even when I hardcode the dropdown options.  

Here I created a plunk, showing the blank option still appears, even when the data is already set to an identical option that appears in the `editDropdownOptionsArray`.   (sorry for extra code, I forked from the plunk above).  

http://plnkr.co/edit/vMLKBcysXI6FsoPb3wHc?p=preview

**Comment by prasannapethe** [comment]
@ArunMNew 
just update the plunker with this you will get the output 

.filter('mapGender', function() {
  var genderHash = {
    'male': 'male',
    'female': 'female'
  };

**Comment by jvquarck** [comment]
@JLLeitschuh had it right, try checking on the console which value is getting the option, since maybe it's "?" instead of "".
`
option:first-child[value="?"]
`

**Comment by AKASH124001** [comment]
I want to hide one specific value from editDropdownOptionsArray on the basis of some condition??
How can I do that??

**Comment by roshenpo007** [comment]
What was the fix for this issue ?  Facing similar defect.

**Comment by jstallm** [comment]
I had the same issue (a blank item showing upon clicking the UI cell. The issue was due to the cell value not being within the defined within the editDropdownOptionsArray.

**Comment by dtripled** [comment]
I figured it out! The css that @JLLeitschuh posted is correct but the first-child value needs to be changed from "" to "?".

So this `#GridCustomId.ui-grid-cell div select option:first-child[value=""] {
    display: none;
  }`

Needs to become: `#GridCustomId.ui-grid-cell div select option:first-child[value="?"] {
    display: none;
  }`

**Comment by gousdem** [comment]
I added those code , but the issue still not fix. Can you help me.

**Comment by Abhilash-Pokhriyal** [comment]
What finally worked for me is:-
#GridCustomId.ui-grid-cell div select option:first-child:empty { display: none; }
